### PR TITLE
Add "method_missing" feature

### PR DESCRIPTION
**Pull request by arBmind**
The "method_missing" method is called for all functions that could not be dispatched.
It get's the name of the function and all the original arguments as arguments.

It's very useful to implement property systems and allow `object.prop` to access the value.

The implementation is very hacky but get's the job done. Very much like the entire dispatch system.

### CODE REVIEW

File: include/chaiscript/dispatchkit/dispatchkit.hpp [hunk 1]
```diff
@@ -782,7 +782,19 @@ namespace chaiscript
...
         {
-          return dispatch::dispatch(get_function(t_name), params, m_conversions);
+          try {
+            return dispatch::dispatch(get_function(t_name), params, m_conversions);
+          }
+          catch(chaiscript::exception::dispatch_error&) {
+            auto functions = get_function("method_missing");
+            if (!functions.empty()) {
+              std::vector<Boxed_Value> tmp_params;
+              tmp_params.push_back(var(t_name));
```
Comment (lefticus) [inline include/chaiscript/dispatchkit/dispatchkit.hpp]: Can you merge these lines into:

```
std::vector<Boxed_Value> tmp_params{var(t_name)};
```

It's *slightly* more efficient, but also more succinct.

**Comment by lefticus** [comment]
@arBmind it has been 12 days since this pull request was last updated.

**Comment by arBmind** [comment]
I did not know that you expect any more work on this.
It works for me, like it is.

**Comment by lefticus** [comment]
@arBmind  That's just my automated build system commenting that a pull request is still open.

**Comment by lefticus** [comment]
@arBmind it has been 7 days since this pull request was last updated.

**Comment by lefticus** [comment]
@arBmind it has been 7 days since this pull request was last updated.

**Pull request merged**